src/Region/RegionHandler.cc

```diff
@@ -221,31 +221,43 @@ void RegionHandler::ImportRegion(int file_id, std::shared_ptr<Frame> frame, CART
 }
 
 void RegionHandler::ExportRegion(int file_id, std::shared_ptr<Frame> frame, CARTA::FileType region_file_type,
-    CARTA::CoordinateType coord_type, std::map<int, CARTA::RegionStyle>& region_styles, std::string& filename,
+    CARTA::CoordinateType coord_type, std::map<int, CARTA::RegionStyle>& region_styles, std::string& filename, bool overwrite,
     CARTA::ExportRegionAck& export_ack) {
     // Export regions to given filename, or return export file contents in ack
     // Check if any regions to export
     if (region_styles.empty()) {
         export_ack.set_success(false);
-        export_ack.set_message("Export failed: no regions requested.");
-        export_ack.add_contents();
+        export_ack.set_message("Export region failed: no regions requested.");
         return;
     }
 
     // Check ability to create export file if filename given
+    std::string message;
     if (!filename.empty()) {
         casacore::File export_file(filename);
+
         if (export_file.exists()) {
-            if (!export_file.isWritable()) {
+            if (!overwrite) {
+                message = "Cannot overwrite existing file.";
                 export_ack.set_success(false);
-                export_ack.set_message("Export region failed: cannot overwrite file.");
-                export_ack.add_contents();
+                export_ack.set_message(message);
                 return;
             }
+
+            if (export_file.isRegular(false)) {
+                if (!export_file.isWritable()) {
+                    message = "Export region failed: cannot overwrite read-only file.";
+                }
+            } else {
+                message = "Export region failed: path is not a file.";
+            }
```

Comment:
I'm not sure about the logic here. Shouldn't this be `isRegular(true)`? But also we need different responses for directories and non-directories. This conditional can't be identical to the save file code because we're using casacore file functions and not filesystem paths (and also the image detection is not applicable), so maybe something like:

* if file exists
  * if it's a directory -> error about directory (never overwritten)
  * if it's a regular file or symlink
    * if no overwrite flag -> error about file or symlink (need confirmation)
    * else if not writeable -> error
  * otherwise -> error about it not being a file
* else if it can't be created... (as before)

Reply:
I was trying to simplify the logic here, since it always fails if overwrite is false (no need to check the type) or if the target is not a file (only need to check type is file).  Currently, the frontend only shows the error and does not respond with a confirmation dialog regardless of the error.  But I will change it if you think additional information would be helpful.